new-lamassu-admin/src/pages/Transactions/CopyToClipboard.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react'
+import classnames from 'classnames'
+import { CopyToClipboard as ReactCopyToClipboard } from 'react-copy-to-clipboard'
+import { replace } from 'lodash/fp'
+import { makeStyles } from '@material-ui/core/styles'
+
+import { cpcStyles } from './Transactions.styles'
+
+import Popover from '../../components/Popper'
+import { ReactComponent as CopyIcon } from '../../styling/icons/action/copy/copy.svg'
+import { comet } from '../../styling/variables'
+
+const CopyToClipboard = ({ className, type, children, ...props }) => {
+  const [anchorEl, setAnchorEl] = useState(null)
+
+  useEffect(() => {
+    if (anchorEl) setTimeout(() => setAnchorEl(null), 3000)
+  }, [anchorEl])
+
+  const useStyles = makeStyles(cpcStyles)
+
+  const classes = useStyles()
+
+  const addrClasses = {
+    [classes.address]: true,
+    [classes.cryptoAddr]: type === 'crypto',
+    [classes.txId]: type === 'txId',
+    [classes.sessionId]: type === 'sessionId'
+  }
+
+  const handleClick = event => {
+    setAnchorEl(anchorEl ? null : event.currentTarget)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
+  const formatAddress = (address = '') => {
+    return address.replace(/(.{5})/g, '$1 ')
+  }
+
+  const formatTxId = (id = '') => {
+    return id.replace(/(.{41})/g, '$1 ')
+  }
+
+  const formatSessionId = (id = '') => {
+    return id.replace(/(.{21})/g, '$1 ')
+  }
```

Comment:
Still not what I meant. This component should do what it says it does, Copy to Clipboard. Formatting should be handled externally. Format address is something that we need, but the other two are just line breaking at the end of the div size, so you should make it break automatically instead of using a function for this.